irg_gazebo_plugins/src/IRGCameraSimPlugins/CameraSimVisualPlugin.cpp

```
@@ -42,7 +42,8 @@ void CameraSimVisualPlugin::Load(rendering::VisualPtr visual, sdf::ElementPtr sd
   m_camera = scene->GetUserCamera(0);
   if (m_camera == nullptr)
   {
-    gzerr << "Camera is null. CameraSimVisualPlugin won't load." << std::endl;
```

This merely silences the error message. Is there a way to only perform this check only within gzclient but not on gzserver? In other words, can you check if the current execution is taking place within the server or the client?

---

I agree it would be preferable to keep the error in case the user camera is somehow missing for the gzclient plug-in. 

`Scene::GetUserCameraCount` could be used to check if there even is a user camera, and therefore whether its running in gzclient or gzserver.

---

Found a different solution that uses Scene::IsServer() to detect if we are running on gzserver.